packages/shadycss/src/scoping-shim.js

```diff
@@ -240,6 +241,29 @@ export default class ScopingShim {
     }
     this._customStyleInterface['enqueued'] = false;
   }
+  /**
+   * Validate ordering of custom styles for Custom Property shim
+   * @param {!Array<!CustomStyleProvider>} customStyles
+   */
+  _validateCustomStyleOrder(customStyles) {
+    const styles = customStyles.map(c => this._customStyleInterface['getStyleForCustomStyle'](c));
+    // sort styles in document order
+    styles.sort((a, b) => {
+      // use `b.compare(a)` to be more straightforward
+      const position = b.compareDocumentPosition(a);
```

Comment:
Are custom styles moved to the main document when using real HTML Imports? IIRC, `compareDocumentPosition` needs the nodes being compared to have the same shallow root node to return a useful value.

Reply:
(from offline) Yes, custom styles move themselves into the main document: https://github.com/Polymer/polymer/blob/master/lib/elements/custom-style.js#L93-L105